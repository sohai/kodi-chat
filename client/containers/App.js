import React from 'react';
import { compose, pure, branch, renderComponent } from 'recompact';
import { PropTypes } from 'prop-types';
import { connect } from 'react-redux';
import Overlay from '../components/Overlay';
import Typography from '../components/Typography';
import styles from './App.css';

const propTypes = PropTypes && {};

const fullPageInfo = info => () => (
  <Overlay>
    <Typography variant="display1" color="light">
      {info}
    </Typography>
  </Overlay>
);

const LoadingOverlay = fullPageInfo('Loading...');

export const AppRender = ({ children, error }) => error!==null ? React.createElement(fullPageInfo(error)) : (
  <div className={styles.root}>{children}</div>
);

export const mapStateToProps = state => ({
  pending: state.connection.pending,
  error: state.connection.error
});

AppRender.propTypes = propTypes;
AppRender.displayName = 'App';
export default compose(
  connect(mapStateToProps),
  branch(({ pending }) => pending, renderComponent(LoadingOverlay)),
  pure
)(AppRender);
