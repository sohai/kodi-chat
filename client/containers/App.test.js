/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';

import createStore from '../store';

import { AppRender, mapStateToProps } from './App';

const store = createStore();

describe('(Container) App', () => {
  const state = store.getState();
  const initialProps = mapStateToProps(state);
  const props = {
    ...initialProps
  };

  it('will receive right props', () => {
    expect(initialProps).toMatchSnapshot();
  });

  it('should render', () => {
    const wrapper = shallow(<AppRender {...props} />, { context: { store } });
    expect(wrapper).toMatchSnapshot();
  });
});
