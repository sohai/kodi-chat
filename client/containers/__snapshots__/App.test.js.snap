// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Container) App should render 1`] = `
<div
  className="root"
/>
`;

exports[`(Container) App will receive right props 1`] = `
Object {
  "error": null,
  "pending": true,
}
`;
